src/components/SubNav/SubNav.tsx

```diff
@@ -0,0 +1,379 @@
+  "section.research-library-lpa.primary",
+  "section.research-library-schomburg.primary",
+  "section.research-library-schwarzman.primary",
+  "section.whats-on.primary",
+  "dark.brand.primary",
+  "dark.section.blogs.primary",
+  "dark.section.books-and-more.primary",
+  "dark.section.connect.primary",
+  "dark.section.education.primary",
+  "dark.section.locations.primary",
+  "dark.section.research.secondary",
+  "dark.section.research-library-lpa.primary",
+  "dark.section.research-library-schomburg.primary",
+  "dark.section.research-library-schwarzman.primary",
+  "dark.section.whats-on.primary",
+];
+export type highlightColors = typeof highlightColorsArray[number];
+
+export interface SubNavProps {
+  /** Additional class name for the `SubNav` component. */
+  className?: string;
+  /**
+   * Primary actions displayed on the left side of the SubNav.
+   * Use SubNavButton and SubNavLink components, which mirror
+   * the DS Button and Link functionalities, plus the isOutlined prop.
+   */
+  primaryActions: ({
+    highlightColor,
+    actionBackgroundColor,
+    selectedItem,
+  }: {
+    highlightColor?: string;
+    actionBackgroundColor?: string;
+    selectedItem?: string;
+  }) => React.ReactNode;
+  /**
+   * Secondary actions displayed on the right side of the SubNav.
+   * Use SubNavButton and SubNavLink components, offering additional
+   * contextual options related to the primary content.
+   */
+  secondaryActions?: ({
+    highlightColor,
+    actionBackgroundColor,
+    selectedItem,
+  }: {
+    highlightColor?: string;
+    actionBackgroundColor?: string;
+    selectedItem?: string;
+  }) => React.ReactNode;
+  /**
+   * The background color to be applied to the hover and active states
+   * of the SubNavLink and SubNavButton components.
+   * This allows for customization of the action items.
+   */
+  actionBackgroundColor?: actionBackgroundColors;
+  /**
+   * Custom color for SubNavLink, SubNavButton, and icons.
+   */
+  highlightColor?: highlightColors;
+  /**
+   * Optional unique ID for accessibility, allowing other components
+   * to reference this element.
+   */
+  id?: string;
+  /** Optional string used to identify and highlight an item. The value should
+   * match the id associated with one of the items.
+   */
+  selectedItem?: string;
+}
+
+export const isIconComponent = (childType: any): boolean => {
+  return (
+    childType.$$typeof === Symbol.for("react.forward_ref") &&
+    childType.displayName === "Icon"
+  );
+};
+
+// Function to check if any of the children is an Icon component
+export const hasIcon = (children: React.ReactNode): boolean => {
+  return React.Children.toArray(children).some((child) => {
+    if (React.isValidElement(child)) {
+      const childType = child.type as React.ComponentType<any>;
+      return isIconComponent(childType);
+    }
+    return false;
+  });
+};
+
+// Function to render content based on the screen size
+export const renderContent = (
```

Comment:
I understand why this was done but I don't think we need `renderContent`, `hasIcon`, or `isIconComponent`. From the spec doc, the best practice and recommendation is that  the _user_ passes in the correct link or button with icon for the appropriate device width. They need to consider this and it's their responsibility and not a responsibility of `SubNav`. So these functions and checks can be removed.